cmake/Codegen.cmake

```diff
@@ -284,6 +285,14 @@ if(INTERN_BUILD_ATEN_OPS)
         endif()
       endif()
       set_source_files_properties(${NEW_IMPL} PROPERTIES COMPILE_FLAGS "${FLAGS} ${EXTRA_FLAGS}")
+    endfunction()
+    foreach(IMPL ${cpu_kernel_cpp_in})
+      file(RELATIVE_PATH NAME "${PROJECT_SOURCE_DIR}/aten/src/ATen/" "${IMPL}")
+      process_vec("${NAME}")
+    endforeach()
+    foreach(IMPL ${cpu_vec_generated_sources})
+      file(RELATIVE_PATH NAME "${CMAKE_BINARY_DIR}/aten/src/ATen/" "${IMPL}")
+      process_vec("${NAME}")
```

Comment:
@peterbell10 I could use some assistance with the cmake logic here. A number of builds are failing due to a race condition in cmake that looks like this:

```
[ 12%] Generating ATen headers
[ 12%] Generating ATen declarations_yaml
make[2]: *** No rule to make target 'aten/src/ATen/UfuncCPUKernel_add.cpp', needed by 'caffe2/CMakeFiles/ATEN_CPU_FILES_GEN_TARGET'.  Stop.
[ 12%] Generating ATen sources
```

(full log https://github.com/pytorch/pytorch/runs/5235004885?check_suite_focus=true )

What appears to be happening is that two things are racing:

1. The `add_custom_command` call for "Generating ATen sources", which would tell cmake about the rule for making "aten/src/ATen/UfuncCPUKernel_add.cpp"
2. The actual compilation process for 'caffe2/CMakeFiles/ATEN_CPU_FILES_GEN_TARGET', which depends on this file.

I am not sure how this race is possible: the ordering seems fine in the source file. Any ideas?